File: data/multilang_comparison/program_julia/program_julia/src/func9.jl
include("program_julia_head.jl")
function func9(vars::JArray_param, PATH0::UInt64, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   array207 = nothing
   if pCounter > 1
       array207 = vars.data[pCounter]
       pCounter -= 1
       array207.refC += 1
   else
       array207 = JArray(zeros(Int, 176), 176, 1, 207)
   end
   params0 = JArray_param{Int}()
   params0.size = 1
   push!(params0.data, array207)
   params0.size += 1
   array208 = func15(params0, loopsFactor);
   if (PATH0 & 1) != 0
      array211 = nothing
      if pCounter > 1
          array211 = vars.data[pCounter]
          pCounter -= 1
          array211.refC += 1
      else
          array211 = JArray(zeros(Int, 818), 818, 1, 211)
      end
      loopLimit61 = (50)/2 + 1
      for i in 1:loopLimit61 
         for i in 0:(array211.size-1) 
            array211.data[i+1] -= 1
         end
         for i in 1:array207.size
             if array207.data[i] == 49
                     return array207
            end
         end
      end
      params1 = JArray_param{Int}()
      params1.size = 3
      push!(params1.data, array207)
      params1.size += 1
      push!(params1.data, array208)
      params1.size += 1
      push!(params1.data, array211)
      params1.size += 1
      array212 = func24(params1, get_path(), loopsFactor);
      array212.refC -= 1
      if array212.refC == 0
          empty!(array212.data)
          array212 = nothing
      end
      array211.refC -= 1
      if array211.refC == 0
          empty!(array211.data)
          array211 = nothing
      end
   
   else 
      params1 = JArray_param{Int}()
      params1.size = 2
      push!(params1.data, array207)
      params1.size += 1
      push!(params1.data, array208)
      params1.size += 1
      array213 = func16(params1, get_path(), loopsFactor);
      array213.refC -= 1
      if array213.refC == 0
          empty!(array213.data)
          array213 = nothing
      end
   end
   array207.refC -= 1
   if array207.refC == 0
       empty!(array207.data)
       array207 = nothing
   end
   return array208;
end

File: data/multilang_comparison/program_go/program_go/src/func9.go
package main
import "math/rand"
var _ = rand.Intn
func Func9(vars *ArrayParam, PATH0 uint64 , loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   var array207 *Array
   if pCounter > 0 {
       array207 = vars.Data[pCounter-1]
       pCounter--
       array207.RefC++
   } else {
       array207 = &Array{
           Size: 176,
           RefC: 1,
           Id: 207,
           Data: make([]uint32, 176),
       }
   }
   var params0 ArrayParam
   params0.Size = 1
   params0.Data = make([]*Array, params0.Size)
   params0.Data[0] = array207
   var array208 *Array = Func15(&params0, loopsFactor);
   params0.Data = nil
   params0.Size = 0
   if (PATH0 & 1) != 0 {
      var array211 *Array
      if pCounter > 0 {
          array211 = vars.Data[pCounter-1]
          pCounter--
          array211.RefC++
      } else {
          array211 = &Array{
              Size: 818,
              RefC: 1,
              Id: 211,
              Data: make([]uint32, 818),
          }
      }
      var loop61 int = 0;
      var loopLimit61 int = (50)/2 + 1;
      for ; loop61 < loopLimit61; loop61++ {
         for i := 0; i < array211.Size; i++ {
             array211.Data[i]--
         }
         for i := 0; i < array207.Size; i++ {
             if array207.Data[i] == 49 {
                 return array207
             }
         }
      }
      var params1 ArrayParam
      params1.Size = 3
      params1.Data = make([]*Array, params1.Size)
      params1.Data[0] = array207
      params1.Data[1] = array208
      params1.Data[2] = array211
      var array212 *Array = Func24(&params1, getPath(), loopsFactor);
      params1.Data = nil
      params1.Size = 0
      array212.RefC--
      if array212.RefC == 0 {
          array212.Data = nil
          array212 = nil
      }
      array211.RefC--
      if array211.RefC == 0 {
          array211.Data = nil
          array211 = nil
      }
   
   } else {
      var params1 ArrayParam
      params1.Size = 2
      params1.Data = make([]*Array, params1.Size)
      params1.Data[0] = array207
      params1.Data[1] = array208
      var array213 *Array = Func16(&params1, getPath(), loopsFactor);
      params1.Data = nil
      params1.Size = 0
      array213.RefC--
      if array213.RefC == 0 {
          array213.Data = nil
          array213 = nil
      }
   }
   array207.RefC--
   if array207.RefC == 0 {
       array207.Data = nil
       array207 = nil
   }
   return array208;
}

